llama_stack/apis/scoring_functions/scoring_functions.py

```diff
@@ -3,33 +3,71 @@
 #
 # This source code is licensed under the terms described in the LICENSE file in
 # the root directory of this source tree.
-
-from typing import Any, Dict, List, Optional, Protocol, runtime_checkable
+from enum import Enum
+from typing import (
+    Any,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Protocol,
+    runtime_checkable,
+    Union,
+)
 
 from llama_models.schema_utils import json_schema_type, webmethod
 from pydantic import BaseModel, Field
+from typing_extensions import Annotated
 
 from llama_stack.apis.common.type_system import ParamType
 
 
-@json_schema_type
-class Parameter(BaseModel):
-    name: str
-    type: ParamType
-    description: Optional[str] = None
-
-
 # Perhaps more structure can be imposed on these functions. Maybe they could be associated
 # with standard metrics so they can be rolled up?
 
 
+@json_schema_type
+class ScoringContextType(Enum):
+    llm_as_judge = "llm_as_judge"
+    answer_parsing = "answer_parsing"
+
+
+@json_schema_type
 class LLMAsJudgeContext(BaseModel):
+    type: Literal[ScoringContextType.llm_as_judge.value] = (
+        ScoringContextType.llm_as_judge.value
+    )
     judge_model: str
     prompt_template: Optional[str] = None
     judge_score_regex: Optional[List[str]] = Field(
         description="Regex to extract the score from the judge response",
         default=None,
     )
+    # TODO: think about whether to put this as a scoring function context or in separate scorer
+    # and how the LLM as judge defines the response
+    judge_grade_metrics: Optional[Dict[str, str]] = Field(
```

Comment:
Alternative Option: we have a separate ScoringFn for SimpleQA task with `SimpleQAContext`.  
```
class SimpleQAContext(BaseModel):
    type: Literal[ScoringContextType.llm_as_judge.value] = (
        ScoringContextType.llm_as_judge.value
    )
    judge_model: str
    prompt_template: Optional[str] = None
    judge_score_regex: 
    judge_grade_metrics: 
    ...
```